### PR TITLE
implementing global exception handling

### PR DESCRIPTION
**Pull request by salinda-yasas-99**

### CODE REVIEW

File: src/main/java/com/gmart/gmart_api/service/UserService.java [hunk 1]
```diff
@@ -49,7 +51,9 @@ public UserDto register(RegisterUserDto userDto) {
         Optional<User> optionalUser = userRepository.findByUsername(userDto.getUsername());
 
         if (optionalUser.isPresent()) {
-            throw new AppException("Login already exists", HttpStatus.BAD_REQUEST);
+           // throw new AppException("Login already exists", HttpStatus.BAD_REQUEST);
```
Comment (nimeshnuwanga) [inline src/main/java/com/gmart/gmart_api/service/UserService.java:54]: remove unnecessary comments

File: src/main/java/com/gmart/gmart_api/exceptions/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        HttpStatus httpStatus = HttpStatus.CONFLICT;
+        var result =CustomException.of(
+                exception.getMessage(),
+                httpStatus,
+                ZonedDateTime.now());
+        return new ResponseEntity<>(result
+                ,httpStatus);
+    }
+
+    //log in exceptions
```
Comment (nimeshnuwanga) [inline src/main/java/com/gmart/gmart_api/exceptions/GlobalExceptionHandler.java:39]: Is this comment needed?

File: src/main/java/com/gmart/gmart_api/exceptions/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.time.ZonedDateTime;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+    // Global exception handler for any other exceptions
```
Comment (nimeshnuwanga) [inline src/main/java/com/gmart/gmart_api/exceptions/GlobalExceptionHandler.java:13]: remove unnecessary comments

File: src/main/java/com/gmart/gmart_api/exceptions/CustomException.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import java.time.format.DateTimeFormatter;
+
+@Data
+@AllArgsConstructor
+public class CustomException {
+    private final String message;
+    private final HttpStatus httpStatus;
+    private final String timeStamp;
+
+    // You can also add a static factory method to build the CustomException with formatted timestamp
```
Comment (nimeshnuwanga) [inline src/main/java/com/gmart/gmart_api/exceptions/CustomException.java:35]: Please remove comments if you are directly copying from the ChatGPT.

File: src/main/java/com/gmart/gmart_api/exceptions/CustomException.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package com.gmart.gmart_api.exceptions;
+//
```
Comment (nimeshnuwanga) [inline src/main/java/com/gmart/gmart_api/exceptions/CustomException.java:2]: remove unnecessary comments

File: src/main/java/com/gmart/gmart_api/controller/UserController.java [hunk 6]
```diff
@@ -33,4 +33,7 @@ public ResponseEntity<String> deleteUserById(@PathVariable String uid){
            return new ResponseEntity<>(result, HttpStatus.OK);
 
     }
+
```
Comment (nimeshnuwanga) [inline src/main/java/com/gmart/gmart_api/controller/UserController.java:36]: Remove unnecessary empty lines

**Review by nimeshnuwanga — CHANGES_REQUESTED** [review]

File: src/main/java/com/gmart/gmart_api/service/UserService.java [hunk 7]
```diff
@@ -62,12 +66,12 @@ public UserDto register(RegisterUserDto userDto) {
...
+                .orElseThrow(() -> new LogInException("Email or Password is incorrect"));
 
         if (passwordEncoder.matches(CharBuffer.wrap(logInDto.getPassword()), user.getPassword())) {
             return toUserDto(user);
         }
-        throw new AppException("Email or Password is incorrect", HttpStatus.BAD_REQUEST);
+        throw new LogInException("Email or Password is incorrect");
     }
 
     //delete user by Id
```
Comment (lakshankd) [inline src/main/java/com/gmart/gmart_api/service/UserService.java:77]: remove this comment as well

File: src/main/java/com/gmart/gmart_api/exceptions/GlobalExceptionHandler.java [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        HttpStatus httpStatus = HttpStatus.NOT_FOUND;
+        var result =CustomException.of(
+                exception.getMessage(),
+                httpStatus,
+                ZonedDateTime.now());
+        return new ResponseEntity<>(result,httpStatus);
+    }
+
+
+
```
Comment (lakshankd) [inline src/main/java/com/gmart/gmart_api/exceptions/GlobalExceptionHandler.java:93]: remove these unnecessary lines

File: src/main/java/com/gmart/gmart_api/exceptions/CustomException.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+//@Data
+//@AllArgsConstructor
+//public class CustomException {
+//
+//    private final String message;
+//    private final HttpStatus httpStatus;
+//    private final ZonedDateTime timeStamp;
+//
+//}
```
Comment (lakshankd) [inline src/main/java/com/gmart/gmart_api/exceptions/CustomException.java:17]: Remove these commented code lines

File: src/main/java/com/gmart/gmart_api/exceptions/GlobalExceptionHandler.java (hunk 3 above)
Comment (salinda-yasas-99) [reply src/main/java/com/gmart/gmart_api/exceptions/GlobalExceptionHandler.java:13]: this comment needed

**Review by lakshankd — APPROVED** [review]

**Review by nimeshnuwanga — APPROVED** [review]

**Pull request merged**